File: public/templates/level2.java
// You have to navigate the robot through more challenging
// maze in this level.
//
// Also, an additional method is available to you:
// - turnRight()
//
// However "turnRight()" just calls "turnLeft()" 3 times.
// Can you make a better "turnRight()"?

import org.jointheleague.ecolban.rpirobot.*;

public class CleverRobot extends IRobotAdapter {
	void go() throws Exception {
		moveForward();
		// Your code here...

	}

	/**
	 * Move the robot forward approximately one "square".
	 */
	void moveForward() throws Exception {
		// Move left and right wheels 500mm/s forward
		driveDirect(500, 500);
		// Let the robot move for 1975 milliseconds
		Thread.sleep(1975);
		// Stop moving
		stop();
	}

	/**
	 * Turns the robot approximately 90 degrees left.
	 */
	void turnLeft() throws Exception {
		// Move left wheel 400mm/s backwards
		// and right wheel 400mm/s forward
		driveDirect(-400, 400);
		// Let the robot move for 436 milliseconds
		Thread.sleep(436);
		// Stop moving
		stop();
	}

	/**
	 * Turns the robot approximately 90 degrees right.
	 */
	void turnRight() throws Exception {
		// Turn left three times
		turnLeft();
		turnLeft();
		turnLeft();
	}

	// DON'T CHANGE ANYTHING BELOW THIS LINE
	CleverRobot(IRobotInterface iRobot) throws Exception {
		super(iRobot);
		go();
	}

	public static void main(String[] args) throws Exception {
		CleverRobot rob = new CleverRobot(new SimpleIRobot());
	}
}

File: public/templates/level2.py
# You have to navigate the robot through more challenging
# maze in this level.
#
# Also, an additional function is available to you:
# - turn_right()
#
# However "turn_right()" just calls "turn_left()" 3 times.
# Can you make a better "turn_right()"?

from time import sleep
from org.jointheleague.ecolban.rpirobot import SimpleIRobot

robot = SimpleIRobot()


def move_forward():
	"""Move the robot forward approximately one square"""
	# Move left and right wheels 500mm/s forward
	robot.driveDirect(500, 500)
	# Let the robot move for 1.975 seconds
	sleep(1.975)
	# Stop moving
	robot.stop()


def turn_left():
	"""Turns the robot approximately 90 degrees left."""
	# Move left wheel 400mm/s backwards
	# and right wheel 400mm/s forward
	robot.driveDirect(-400, 400)
	# Let the robot move for 0.436 seconds
	sleep(0.436)
	# Stop moving
	robot.stop()


def turn_right():
	"""Turns the robot approximately 90 degrees right."""
	# Turn left three times
	turn_left()
	turn_left()
	turn_left()


move_forward()
# Your code here...
